pkg/adapter/content.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/containerd/containerd/metadata"
+	"github.com/dustin/go-humanize"
+	"github.com/goharbor/acceleration-service/pkg/config"
+	"github.com/sirupsen/logrus"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	bucketKeyVersion       = []byte("v1")
+	bucketKeyObjectContent = []byte("content")
+	bucketKeyObjectBlob    = []byte("blob")
+
+	bucketKeySize = []byte("size")
+)
+
+type Content struct {
+	db *metadata.DB
+}
+
+func NewContent(db *metadata.DB) *Content {
+	return &Content{
+		db: db,
+	}
+}
+
+// Size return the size of local caches size
+func (content *Content) Size() (int64, error) {
+	var contentSize int64
+	if err := content.db.View(func(tx *bolt.Tx) error {
+		bucket := getBucket(tx, bucketKeyVersion, []byte("acceleration-service"), bucketKeyObjectContent, bucketKeyObjectBlob)
+		// if can't find blob bucket, it maens content store is empty
+		if bucket == nil {
+			return nil
+		}
+		return bucket.ForEach(func(key, value []byte) error {
+			if subBucket := bucket.Bucket(key); subBucket != nil {
+				size, err := blobSize(subBucket)
+				if err != nil {
+					return err
+				}
+				contentSize += size
+			}
+			return nil
+		})
+	}); err != nil {
+		return 0, err
+	}
+	return contentSize, nil
+}
+
+// GC clean the local caches by cfg.Provider.GCPolicy configuration
+func (content *Content) GC(ctx context.Context, cfg *config.Config) error {
```

Comment:
Maybe pass the `cfg.Provider.GCPolicy` to `NewContent`, to avoid invoking `humanize.ParseBytes` repeatedly.